vault/acl.go

```diff
@@ -318,6 +318,13 @@ func (a *ACL) Capabilities(ctx context.Context, path string) (pathCapabilities [
 func (a *ACL) AllowOperation(ctx context.Context, req *logical.Request, capCheckOnly bool) (ret *ACLResults) {
 	ret = new(ACLResults)
 
+	// check if format of kv request is a v1 or v2 request:
```

Comment:
I don't think we should put secret-engine-specific code into generic core code like acl.go.